src/components/Datatable/Datatable.vue

```diff
@@ -0,0 +1,280 @@
+          >
+            <slot name="column" :column="column">
+              <slot
+                :name="'column-' + columns[columnIndex].value"
+                :column="column"
+              >
+                {{ columns[columnIndex].text }}
+              </slot>
+
+              <i
+                v-if="currentSortDir == 'asc'"
+                class="pi pi-arrow-up-line pi-1x"
+              />
+              <i
+                v-else-if="currentSortDir == 'desc'"
+                class="pi pi-arrow-down-line pi-1x"
+              />
+
+              <div v-else class="rotate-arrow">
+                <i class="pi pi-code-line pi-1x" />
+              </div>
+            </slot>
+          </div>
+        </th>
+      </tr>
+      <tr class="table-row">
+        <td :colspan="Object.keys(columns).length" class="search-input">
+          <Input v-model="search" placeholder="Search...">
+            <template v-slot:prepend>
+              <div class="search-icon">
+                <i class="pi pi-search-line pi-1x" />
+              </div>
+            </template>
+          </Input>
+        </td>
+      </tr>
+    </thead>
+
+    <tbody class="table-body">
+      <tr v-for="(row, rowIndex) in sortedColumns" :key="rowIndex">
+        <td
+          v-for="(column, columnIndex) in columns"
+          :key="columnIndex"
+          :align="column.align || 'start'"
+          class="table-cell"
+        >
+          <slot name="item" :item="row">
+            <slot :name="'item-' + Object.keys(row)[columnIndex]" :item="row">
+              {{ row[Object.keys(row)[columnIndex]] }}
+            </slot>
+          </slot>
+        </td>
+      </tr>
+
+      <tr>
+        <td
+          v-if="sortedColumns.length === 0"
+          :colspan="Object.keys(columns).length"
+          align="center"
+          class="no-search-results"
+        >
+          <div>No results found</div>
+          <Button @click="clearSearch" miter width="170px" height="36px"
+            >Clear search</Button
+          >
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import Button from '../Button/Button.vue'
+import Input from '../Input/Input.vue'
+
+export default defineComponent({
+  name: 'Datatable',
+  components: { Button, Input },
+  props: {
+    columns: { type: Array, required: true, default: () => [] },
+    items: { type: Array, required: true, default: () => [] },
+    dir: { type: String, required: false, default: '' },
+    sortBy: { type: String, required: false, default: 'name' }
+  },
+  emits: ['update:dir'],
+  mounted() {
+    window.addEventListener('resize', this.handleResize)
+  },
+  beforeDestroy() {
+    window.removeEventListener('resize', this.handleResize)
+  },
+  data() {
+    return {
+      currentSortDir: '' as String,
+      search: '' as String,
+      sortedItems: [] as object[],
+      windowWidth: window.innerWidth as Number
```

Comment:
I think we should do the responsive logic with css instead of js. We're only hiding a handful of dom elements on mobile, the cost of using js to hide them probably outweighs the benefits of `v-if` over `display: none;`